MoreLinq/ListLike.cs

```diff
@@ -21,34 +21,35 @@ namespace MoreLinq
     using System.Collections.Generic;
 
     /// <summary>
-    /// Represents an list-like (indexable) data structure.
+    /// Represents a union over list types implementing either
```

Comment:
Instead of a union, what about a delegating tuple?

```csharp
static class ListLike 
{
  public static (Func<int> GetCount, Func<int, T> GetItem) => (() => list.Count, (i) => list[i]);
  public static (Func<int> GetCount, Func<int, T> GetItem) => (() => list.Count, (i) => list[i]);
}
```

Reply:
I don't this is syntactically correct, but I think I get your idea nonetheless. This would cause two closures and indirect calls so have my doubts it's as optimal.